packages/nimble-components/src/rich-text/models/markdown-parser.ts

```diff
@@ -65,16 +132,101 @@ export class RichTextMarkdownParser {
+            }
+        });
+    }
+
+    private static getDisplayName(
+        mentionHref: string,
+        mention: MentionInternals<RichTextMentionConfig>
+    ): string | undefined {
+        const mentionMapping = mention?.mentionConfig?.mappingConfigs.get(mentionHref);
+        return mentionMapping?.displayName;
     }
 
-    private static getSchemaWithLinkConfiguration(): Schema {
+    private static extractMentionId(
+        mention: MentionInternals<RichTextMentionConfig>,
+        mentionHref: string
+    ): string {
+        if (!mention.validConfiguration) {
+            return '';
+        }
+
+        const extractIdRegex = new RegExp(mention.pattern ?? '');
+        const regexpArray = extractIdRegex.exec(mentionHref);
+
+        // Matches and gets the first group specified in the regex pattern
+        // that renders as an alternative to the display name if missing.
+        return regexpArray?.[1] ?? '';
+    }
+
+    // Validates the mention Href in the markdown string with the input configuration values
+    private static validateMentionPattern(
+        mention: MentionInternals<RichTextMentionConfig>,
+        mentionHref: string
+    ): boolean {
+        const regexPattern = new RegExp(mention.pattern ?? '');
+        return regexPattern.test(mentionHref);
+    }
+
+    private static getMentionHref(state: StateInline): string | undefined {
+        const max = state.posMax;
+        const openingTag = 0x3c; // '<'
+        const closingTag = 0x3e; // '>'
+        let position: number;
+
+        if (state.src.charCodeAt(state.pos) !== openingTag) {
+            return undefined;
+        }
+
+        position = state.pos + 1;
+
+        while (
+            position < max
+            && state.src.charCodeAt(position) !== closingTag
+        ) {
+            position += 1;
+        }
+
+        if (position === max && state.src.charCodeAt(position) !== closingTag) {
+            return undefined;
+        }
+
+        return state.src.slice(state.pos + 1, position);
+    }
+
+    private static getCustomSchemaConfiguration(): Schema {
         return new Schema({
-            nodes: schema.spec.nodes,
+            nodes: schema.spec.nodes.addToEnd('mention', {
+                attrs: {
+                    mentionHref: { default: '' },
+                    mentionLabel: { default: '' },
+                    disableEditing: { default: 'true' }
+                },
+                group: 'inline',
+                inline: true,
+                content: 'inline*',
+                toDOM(node) {
+                    const { mentionHref, mentionLabel, disableEditing } = node.attrs;
+                    return [
+                        RichTextMarkdownParser.viewElement,
+                        {
+                            'mention-href': mentionHref as string,
+                            'mention-label': mentionLabel as string,
+                            'disable-editing': disableEditing as string
```

Comment:
I think I understand the gist of your response, but my concern was that `disable-editing` is a boolean attribute, and we seem to be assigning a string value. I've looked at the documentation for this and I couldn't determine whether we can instead set a boolean value, i.e.:
```TS
attrs: {
    mentionHref: { default: '' },
    mentionLabel: { default: '' },
    disableEditing: { default: true },
    viewElement: { default: '' }
},
...
toDOM(node) {
    ...
    return [
        viewElement,
        {
              'mention-href': mentionHref as string,
              'mention-label': mentionLabel as string,
              'disable-editing': disableEditing as boolean
         }
```
Or if we always want the view element to have `disable-editing` set here, it would be clearer to do:
 ```TS
    return [
        viewElement,
        {
              'mention-href': mentionHref as string,
              'mention-label': mentionLabel as string,
              'disable-editing': 'true'
         }
```
I'm just concerned that the way it is currently written, it looks like if `disableEditing` was the string `"false"`, we would enable editing on that view element, but that is not true.

Reply:
Sorry, I misunderstood. As you suggested, the second approach is clearer as it will always be in view mode, so I have implemented the same.